ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -177,11 +177,6 @@ private void initReplDump(ASTNode ast) throws HiveException {
     for (String dbName : Utils.matchesDb(db, dbNameOrPattern)) {
       Database database = db.getDatabase(dbName);
       if (database != null) {
-        if (!isMetaDataOnly && !ReplChangeManager.isSourceOfReplication(database)) {
```

Comment:
Is a check needed on the load side for upgrade scenarios? Say source cluster is not upgraded but the target is.

Reply:
In case the source is old, it won't allow dump without the conf set, if it is new, we would add the config, so in all cases the conf would be set, so we won't need this check.